crates/build-utils/src/docker.rs

```diff
@@ -0,0 +1,78 @@
+use std::{
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+use thiserror::Error;
+use tracing::info;
+
+pub fn build_image(dockerfile_workspace_relative_path: &str, tag: &str) -> Result<(), Error> {
+    // Check that Docker is installed and available
+    if Command::new("docker")
+        .arg("--version")
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()
+        .is_err()
+    {
+        return Err(Error::DockerIsNotAvailable);
+    }
+
+    let cargo_workspace_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("../..")
```

Comment:
When `ere` is used as a dependency, `CARGO_MANIFEST_DIR` will point to the path where the crate was downloaded by `cargo`. Then `../..` put us in the workspace place.

I think this works since the crate is pulled as a git repo? I'm not entirely sure.
If at some point we end up publishing crates in crate.io, we can figure out. If this doesn't work, we can embed the contents of Dockerfiles at compile time with `include_str!` and write stuff in temp files for Docker or similar.